source/PlayerInfo.cpp

```diff
@@ -851,20 +854,28 @@ map<const shared_ptr<Ship>, vector<string>> PlayerInfo::FlightCheck() const
 			}
 			// This ship requires a bay to travel between systems.
 			else if(bayType.second > 0)
-				--bayType.second;
-			else
 			{
-				// Include the lack of bay availability amongst any other
-				// warnings for this carriable ship.
-				auto it = flightChecks.find(carriable);
-				string warning = "no bays?";
-				if(it != flightChecks.end())
-					it->second.emplace_back(warning);
-				else
-					flightChecks.emplace(carriable, vector<string>{warning});
+				--bayType.second;
+				dockedShips[carriable] = true;
 			}
+			else
+				dockedShips[carriable] = false;
 		}
 	}
+
+	for(const auto &ship : ships)
+		if(ship->GetSystem() == system && !ship->IsDisabled() && !ship->IsParked())
+		{
+			// Check if ship is docked
+			auto it = dockedShips.find(ship);
+			auto checks = ship->FlightCheck(it != dockedShips.end() && it->second);
+			// Include the lack of bay availability amongst any other
+			// warnings for this carriable ship.
+			if(it != dockedShips.end() && !it->second)
+				checks.emplace_back("no bays?");
+			if(!checks.empty())
+				flightChecks.emplace(ship, checks);
```

Comment:
May as well save it in a bool and name it (more for clarity than for optimization, really)
```suggestion
			const bool isCarried = it != dockedShips.end();
			auto checks = ship->FlightCheck(isCarried && it->second);
			// Include the lack of bay availability amongst any other
			// warnings for this carriable ship.
			if(isCarried && !it->second)
				checks.emplace_back("no bays?");
			if(!checks.empty())
				flightChecks.emplace(ship, checks);
```